src/MSALWrapper.Test/GlobalLockTest.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Authentication.MSALWrapper;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// The <see cref="GlobalLockTest"/> test.
+    /// </summary>
+    internal class GlobalLockTest
+    {
+        /// <summary>
+        /// The number of threads or tasks.
+        /// </summary>
+        private const int NumberOfThreads = 10;
+
+        /// <summary>
+        /// The simulated lockname.
+        /// </summary>
+        private string lockName;
+
+        /// <summary>
+        /// The setup.
+        /// </summary>
+        [SetUp]
+        public void Setup()
+        {
+            this.lockName = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Test mutex in single task.
+        /// </summary>
+        [Test]
+        public void TestSingleMutex()
+        {
+            int sum = 0;
+            Task.Run(() =>
+            {
+                using (new GlobalLock(this.lockName))
+                {
+                    sum++;
+                }
+            }).Wait();
+            Assert.AreEqual(1, sum);
+        }
+
+        /// <summary>
+        /// Test mutex in tasks.
+        /// </summary>
+        [Test]
+        public void TestMutexInTasks()
+        {
+            int cas = 0;
+            int sum = 0;
+            var tasks = new List<Task>();
+            for (int i = 0; i < NumberOfThreads; i++)
+            {
+                var task = new Task(() =>
+                {
+                    using (new GlobalLock(this.lockName))
+                    {
+                        if (Interlocked.CompareExchange(ref cas, 1, 0) == 1)

```

Comment:
This is a pretty non-standard API to be using (unless you are someone who is used to writing parallel or concurrent code - which generally most people are not). So I think this is a good opportunity for a comment to help describe what is being checked here, and also potentially at the start of the method to describe the overall goal of this test. What is this test asserting, and how is it doing so?

Does this test fail without the logic in GlobalLock?